### DISCUSSION TITLE
Есть ли в Maxsite автозаполнение Meta-description и title

### DISCUSSION THREAD

**Discussion by nvda-ru** [maxsite/cms#472] [Q&A] [answered]
Всем доброго времени суток!

Подскажите пожалуйста, есть лив Maxsite под капотом функционал для автоматического заполнения Meta-description и title, если они не заполнены вручную.
Может под капотом есть возможность это включить, кому это необходимо, но настройка не выведена в админ панель.
Много лет на социальном проекте использую плагин SeoGen
http://maxhub.ru/page/plugin-seogen

Но плагин не желает работать на последних версиях Maxsite и версии php выше 7.1.
Сейчас использую MS 107.4 и php7.1, но есть желание обновить версию сайта и поднять php до 8.0.
При этом хочется сохранить функционал автозаполнения этих Meta. Вручную заполнять их не хочется, потому что уникальности это не требует, а для поисковых систем достаточно и того что заполняется автоматически.

Возможно подобные опции не всем нужны, поэтому плагин давно не обновляли, а под капотом MS может что-то есть для этого.
А может, есть что-то по проще, для добавления в шаблон, примерно как это можно было делать в WP в файле функций.

Благодарю всех за информацию и помощь!

**Comment by maxsite** [comment, marked as answer]
Так они и так автоматом заполняются из заголовка и текста записи.

**Comment by nvda-ru** [comment]
Много лет использую MS, но не пробовал без этого плагина.
Думал, раз в плагине есть эти функции для автозаполнения meta-description и title, то этого нет в самом Maxsite.
Попробую в ближайшие дни, может действительно без плагина будет как надо.
И я столько лет пользовался этим плагином и не знал, что в MS это всё давно есть.
Максим, а может в MS есть возможность изменять ссылки типа:
https://domain.com/page/hi-friend
 на
https://domain.com/hi-friend
Или это можно делать только плагином Ильи Земского?
Этот плагин хороший, но мне не нужно от него столько много функционала.
Вдруг это давно есть под капотом MS, а я просто не знаю об этом.
Когда то стал использовать этот плагин, потому что хотелось иметь более удобную ссылку без page, для использования в незрячей среде, но при обновлении MS остаёшся в зависимости от этого плагина.

**Comment by maxsite** [comment]
Это много раз уже обсуждали. Сменить ЧПУ не получится. Можно сделать редиректы или синонимы страниц без «page», но полностью избавиться не получится.

**Comment by nvda-ru** [comment]
Благодарю. Всё понял.
Значит буду дальше использовать плагин, раз когда-то начал делать с ним.
А про синонимы ещё раз почитаю в документации, возможно что-то упустил.

**Comment by nvda-ru** [comment]
Здравствуйте. Вчера, делал публикацию новости, но перед этим отключил плагин SeoGen.
После, сохранения записи, перешел во вкладку:
"Основные опции:
но вот эти два поля остались не заполнеными: 
Титул страницы (title)
Описание страницы (meta description)


Не могу понять, почему эти два поля не заполнились.

Может плагин нужно вовсе удалить, отключить не достаточно, и он всё равно что-то блокирует.

**Comment by maxsite** [comment]
Когда эти поля пустые, то title и meta description формируются автоматом. Посмотрите исходный код станицы Ctrl+U.

**Comment by nvda-ru** [comment]
Действительно, в код не смотрел.
Meta-description после добавления новой записи оказался заполнен.
Меня ввело в заблуждение, то что после сохранения записи, указанные два поля остались пустыми.
Когда заглянул в код, то обратил внимание, что размер текста в meta-Discription по количеству символов больше, чем это было при использовании SeoGen.
Туда попадает часть текста, которая после [cut].
Подскажите пожалуйста, в каком файле можно найти настройку, где можно отрегулировать количество символов для meta-discription?
Благодарю.

**Comment by maxsite** [comment]
В опциях — Формировать описание (description) из текста записи

**Comment by nvda-ru** [comment]
Благодарю.
Даже и не знал, что в общих настройках, вкладка мета-теги, есть такая функция.
"
Если описание записи пусто, то разрешить использовать начало текста записи для формирования description. Указывается число слов. Если указать 0, то опция будет проигнорирована.
"

Даже не помню когда туда в последний раз заходил.
Хотя, всегда читаю ваши посты, но видимо где-то прослушал про эту опцию или банально запамятовал.
